File: Practica 2 aux/Composicion y Agregacion/Ejercicio 7/Python/Estudiante.py
class Estudiante:
    def __init__(self, nombre, carrera, semestre):
        self.nombre = nombre
        self.carrera = carrera
        self.semestre = semestre
    def mostrar(self):
        print("Nombre:", self.nombre)
        print("Carrera:", self.carrera)
        print("Semestre:", self.semestre)
class Universidad:
    def __init__(self, nombre):
        self.nombre = nombre
        self.estudiantes = []
    def agregar(self, estudiante):
        self.estudiantes.append(estudiante)
    def mostrar(self):
        print("Universidad:", self.nombre)
        for e in self.estudiantes:
            e.mostrar()
u =  Universidad("UPB");
u.agregar(Estudiante("Carlos", "Ingeniería", 3));
u.agregar(Estudiante("María", "Arquitectura", 1));
u.mostrar();

File: Practica 2 aux/Composicion y Agregacion/Ejercicio 7/Java/Estudiante.java
import java.util.ArrayList;
class Estudiante {
    public String nombre;
    public String carrera;
    public int semestre;
    public Estudiante(String nombre, String carrera, int semestre) {
        this.nombre = nombre;
        this.carrera = carrera;
        this.semestre = semestre;
    }
    public void mostrar() {
        System.out.println("Estudiante: " + nombre + ", Carrera: " + carrera + ", Semestre: " + semestre);
    }
}
class Universidad {
    public String nombre;
    public ArrayList<Estudiante> estudiantes;
    public Universidad(String nombre) {
        this.nombre = nombre;
        this.estudiantes = new ArrayList<>();
    }
    public void agregar(Estudiante e) {
        estudiantes.add(e);
    }
    public void mostrar() {
        System.out.println("Universidad: " + nombre);
        for (Estudiante e : estudiantes) {
            e.mostrar();
        }
    }
}
public class Ej7 {
    public static void main(String[] args) {
        Universidad u = new Universidad("Univalle");
        Estudiante e1 = new Estudiante("Ana", "Ingeniería", 3);
        Estudiante e2 = new Estudiante("Luis", "Medicina", 5);
        u.agregar(e1);
        u.agregar(e2);
        u.mostrar();
    }
}
